EventFilter/CSCRawToDigi/src/CSCDigiToRaw.cc

```diff
@@ -161,28 +162,30 @@ void CSCDigiToRaw::add(const CSCStripDigiCollection& stripDigis,
       for (; digiItr != last; ++digiItr) {
         CSCStripDigi digi = *digiItr;
         int strip = digi.getStrip();
+        // From LS1 on ME1a strips are unganged
         if (fedInfo.formatVersion_ == 2013) {
           if (me1a && zplus) {
-            digi.setStrip(49 - strip);  // 1-48 -> 48-1
+            digi.setStrip(CSCConstants::NUM_STRIPS_ME1A_UNGANGED + 1 - strip);  // 1-48 -> 48-1
           }
           if (me1b && !zplus) {
-            digi.setStrip(65 - strip);  // 1-64 -> 64-1
+            digi.setStrip(CSCConstants::NUM_STRIPS_ME1B + 1 - strip);  // 1-64 -> 64-1
           }
           if (me1a) {
```

Comment:
I am not sure what to make of this 
```python
        // From LS1 on ME1a strips are unganged                                                                                                                                    
        if (fedInfo.formatVersion_ == 2013) {
          if (me1a && zplus) {
            digi.setStrip(CSCConstants::NUM_STRIPS_ME1A_UNGANGED + 1 - strip);  // 1-48 -> 48-1                                                                                    
          }
          if (me1b && !zplus) {
            digi.setStrip(CSCConstants::NUM_STRIPS_ME1B + 1 - strip);  // 1-64 -> 64-1                                                                                             
          }
          if (me1a) {
            strip = digi.getStrip();  // reset back 1-16 to 65-80 digi                                                                                                             
            digi.setStrip(strip + CSCConstants::NUM_STRIPS_ME1B);
          }
        }
        // During Run-1 ME1a strips are triple-ganged                                                                                                                              
        else {
          if (me1a && zplus) {
            digi.setStrip(CSCConstants::NUM_STRIPS_ME1A_GANGED + 1 - strip);  // 1-16 -> 16-1                                                                                      
          }
          if (me1b && !zplus) {
            digi.setStrip(CSCConstants::NUM_STRIPS_ME1B + 1 - strip);  // 1-64 -> 64-1                                                                                             
          }
          if (me1a) {
            strip = digi.getStrip();  // reset back 1-16 to 65-80 digi                                                                                                             
            digi.setStrip(strip + CSCConstants::NUM_STRIPS_ME1B);
          }
        }
```
@barvic Can you clarify the intention of this block? In particular why is there ```if (me1a && zplus) {``` and ```if (me1a) {``` separately?

Reply:
@perrotta @ptcox Then the following suggestion would make it clearer:
```c++
        // For ME1/b, the switching is the same for all Runs
        if (me1b && !zplus) {
          digi.setStrip(CSCConstants::NUM_STRIPS_ME1B + 1 - strip);  // 1-64 -> 64-1
        }

        // For ME1/a, the switching is dependent on the Run.
        if (me1a) {
          if (zplus) {
            // Run-2: unganged strips
            if (fedInfo.formatVersion_ == 2013) {
              digi.setStrip(CSCConstants::NUM_STRIPS_ME1A_UNGANGED + 1 - strip);  // 1-48 -> 48-1
            }
            // Run-1: triple-ganged strips
            else {
              digi.setStrip(CSCConstants::NUM_STRIPS_ME1A_GANGED + 1 - strip);  // 1-16 -> 16-1
            }
          }
          /* Now add 64 so that ME1/a extends ME1/b
             - Run-1: 80-65 (+z), 65->80 (-z)
             - Run-2: 96-65 (+z), 65->96 (-z)
          */
          digi.setStrip(digi.getStrip() + CSCConstants::NUM_STRIPS_ME1B);
        }
```

Reply:
See https://github.com/cms-sw/cmssw/pull/33429/commits/211bcbd66921375a7eb3e18e81eb9f20b6bc4e4c